RestaurantReservation.Db/Repositories/EmployeeRepository.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.DataModels;
+using RestaurantReservation.Db.Enums;
+using RestaurantReservation.Db.IServices;
+using RestaurantReservation.Db.Utilities;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class EmployeeRepository : IRepositoryServices<Employee, string>, IEmployeeServices
+    {
+        private readonly RestaurantReservationDbContext _dbContext;
+
+        public EmployeeRepository(RestaurantReservationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<OperationResult<string>> AddAsync(Employee employee)
+        {
+            var existingRestaurant = await _dbContext.Restaurants.FindAsync(employee.RestaurantId);
+
+            if (existingRestaurant == null)
+            {
+                return OperationResult<string>.FailureResult("Invalid RestaurantId. The associated restaurant does not exist.");
+            }
+
+            await _dbContext.Employees.AddAsync(employee);
+            await _dbContext.SaveChangesAsync();
+            return OperationResult<string>.SuccessResult($"id: {employee.Id}");
+        }
+
+        public async Task<bool> DeleteAsync(int id)
```

Comment:
you return Operational result in some cases and not in other, any reason or specific pattern when you use it ?